skimage/_shared/testing.py

```diff
@@ -346,19 +352,24 @@ def run_in_parallel(num_threads=2, warnings_matching=None):
     def wrapper(func):
         @functools.wraps(func)
         def inner(*args, **kwargs):
-            with expected_warnings(warnings_matching):
-                threads = []
-                for i in range(num_threads - 1):
-                    thread = threading.Thread(target=func, args=args, kwargs=kwargs)
-                    threads.append(thread)
-                for thread in threads:
-                    thread.start()
-
+            if not is_wasm:
+                with expected_warnings(warnings_matching):
+                    threads = []
+                    for i in range(num_threads - 1):
+                        import threading
+
+                        thread = threading.Thread(target=func, args=args, kwargs=kwargs)
```

Comment:
I have shifted the threading import to inside the function – this function is not a part of the public API offered by `scikit-image`, but if needed I can make it a top-level import again and skip it on WASM.

Reply:
Thanks, that will indeed be better. Addressed in 0b1c933ca36e3f12810dd7ed27f870457a76e8df.

Reply:
Oh, I meant to return even earlier and skip the definition of `inner`, see https://github.com/scikit-image/scikit-image/pull/7350/commits/e94b1cd15082f5ff2afc0ef0dcabae9cb6f100a2.